### ISSUE TITLE
Flaky test: ClosingConnectionMidChunkPrefixThrows

### ISSUE THREAD

**Issue by ryanbrandenburg** [aspnet/KestrelHttpServer#2766]
This test [fails](http://aspnetci/viewLog.html?buildId=517539&buildTypeId=XPlat_Windows_Win10Universe) occasionally with the following error:
```
System.Threading.Tasks.TaskCanceledException : The request was aborted
---- Microsoft.AspNetCore.Connections.ConnectionAbortedException : The connection was aborted
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpRequestStream.<CopyToAsyncInternal>d__23.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.AspNetCore.Server.Kestrel.FunctionalTests.ChunkedRequestTests.<>c__DisplayClass15_0.<<ClosingConnectionMidChunkPrefixThrows>b__0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.AspNetCore.Testing.TaskExtensions.<TimeoutAfter>d__0`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.AspNetCore.Server.Kestrel.FunctionalTests.ChunkedRequestTests.<ClosingConnectionMidChunkPrefixThrows>d__15.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
----- Inner Stack Trace -----
   at System.IO.Pipelines.PipeCompletion.ThrowLatchedException()
   at System.IO.Pipelines.Pipe.GetReadResult(ReadResult& result)
   at System.IO.Pipelines.Pipe.GetReadAsyncResult()
   at System.IO.Pipelines.Pipe.DefaultPipeReader.GetResult(Int16 token)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Adapter.Internal.RawStream.<ReadAsyncInternal>d__22.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Adapter.Internal.AdaptedPipeline.<ReadInputAsync>d__18.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.IO.Pipelines.PipeCompletion.ThrowLatchedException()
   at System.IO.Pipelines.Pipe.GetReadResult(ReadResult& result)
   at System.IO.Pipelines.Pipe.GetReadAsyncResult()
   at System.IO.Pipelines.Pipe.DefaultPipeReader.GetResult(Int16 token)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.Http1MessageBody.<PumpAsync>d__4.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.IO.Pipelines.PipeCompletion.ThrowLatchedException()
   at System.IO.Pipelines.Pipe.GetReadResult(ReadResult& result)
   at System.IO.Pipelines.Pipe.GetReadAsyncResult()
   at System.IO.Pipelines.Pipe.DefaultPipeReader.GetResult(Int16 token)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.MessageBody.<CopyToAsync>d__22.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpRequestStream.<CopyToAsyncInternal>d__23.MoveNext()
------- Stdout: -------
| [0.002s] TestLifetime Information: Starting test ClosingConnectionMidChunkPrefixThrows-http://127.0.0.1:0 at 2018-07-28T02:49:25
| [0.003s] Microsoft.AspNetCore.Hosting.Internal.WebHost Debug: Hosting starting
| [0.059s] Microsoft.AspNetCore.Hosting.Internal.WebHost Debug: Hosting started
| [0.059s] Microsoft.AspNetCore.Hosting.Internal.WebHost Debug: Loaded hosting startup assembly Libuv.FunctionalTests, Version=3.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60
| [0.073s] Microsoft.AspNetCore.Server.Kestrel Debug: Connection id "0HLFK6IT8SMST" started.
| [0.077s] Microsoft.AspNetCore.Hosting.Internal.WebHost Information: Request starting HTTP/1.1 POST http:///  
| [0.078s] Microsoft.AspNetCore.Server.Kestrel Debug: Connection id "0HLFK6IT8SMST", Request id "0HLFK6IT8SMST:00000001": started reading request body.
| [0.078s] Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv Debug: Connection id "0HLFK6IT8SMST" received FIN.
| [0.078s] Microsoft.AspNetCore.Server.Kestrel Debug: Connection id "0HLFK6IT8SMST" disconnecting.
| [0.078s] Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv Debug: Connection id "0HLFK6IT8SMST" sending FIN.
| [0.080s] Microsoft.AspNetCore.Server.Kestrel Debug: Connection id "0HLFK6IT8SMST", Request id "0HLFK6IT8SMST:00000001": done reading request body.
| [0.080s] Microsoft.AspNetCore.Hosting.Internal.WebHost Information: Request finished in 2.9645ms 0 
| [0.080s] Microsoft.AspNetCore.Server.Kestrel Debug: Connection id "0HLFK6IT8SMST" stopped.
| [0.081s] Microsoft.AspNetCore.Hosting.Internal.WebHost Debug: Hosting shutdown
| [0.124s] TestLifetime Information: Finished test ClosingConnectionMidChunkPrefixThrows-http://127.0.0.1:0 in 0
...
```
Other tests within that build may have failed with a similar message, but they are not listed here. Check the link above for more info.

CC @Eilon,@muratg,@mkArtakMSFT

This issue was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
Please use this workflow to address this flaky test issue, including checking applicable checkboxes and filling in the applicable "TODO" entries:

* Is this actually a flaky test?
  * No, this is a regular test failure, fix the test/product (TODO: Link to commit/PR)
  * Yes, proceed below...

* Is this test failure caused by product code flakiness? (Either this product, or another product this test depends on.)
  * [ ] File a bug against the product (TODO: Link to other bug)
  * Is it possible to change the test to avoid the flakiness?
    * Yes? Go to the "Change the test!" section.
    * No?
      * [ ] Disable the test (TODO: Link to PR/commit)
      * [ ] Wait for other bug to be resolved
      * [ ] Wait for us to get build that has the fix
      * [ ] Re-enable our test (TODO: Link to PR/commit)
      * [ ] Close this bug

* Is it that the test itself is flaky? This includes external transient problems (e.g. remote server problems, file system race condition, etc.)
  * Is there is a way to change our test to avoid this flakiness?
    * Yes? Change the test!
      * [ ] Change the test to avoid the flakiness, for example by using a different test strategy, or by adding retries w/ timeouts (TODO: Link to PR/commit)
      * [ ] Run the test 100 times locally as a sanity check.
      * [ ] Close this bug
    * No?
      * [ ] Delete the test because flaky tests are not useful (TODO: Link to PR/commit)

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
ClosingConnectionMidChunkPrefixThrows [failed again](http://aspnetci/viewLog.html?buildId=517537&buildTypeId=XPlat_Windows_Win8_Universe).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
DoesNotThrowOnFin [failed again](http://aspnetci/viewLog.html?buildId=517997&buildTypeId=XPlat_Mac_MacUniverseCoreCLR).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
DoesNotThrowOnFin [failed again](http://aspnetci/viewLog.html?buildId=518390&buildTypeId=XPlat_Mac_MacUniverseCoreCLR).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
DoesNotThrowOnFin [failed again](http://aspnetci/viewLog.html?buildId=518377&buildTypeId=XPlat_HighSierraOsxUniverse).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
ClosingConnectionMidChunkPrefixThrows [failed again](http://aspnetci/viewLog.html?buildId=518392&buildTypeId=XPlat_Windows_Win8_Universe).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
ClosingConnectionMidChunkPrefixThrows [failed again](http://aspnetci/viewLog.html?buildId=518375&buildTypeId=XPlat_Windows_Win7_Universe).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
DoesNotThrowOnFin [failed again](http://aspnetci/viewLog.html?buildId=518373&buildTypeId=XPlat_Windows_Win10Universe).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
ClosingConnectionMidChunkPrefixThrows [failed again](http://aspnetci/viewLog.html?buildId=518450&buildTypeId=Lite_UniverseTest).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
ClosingConnectionMidChunkPrefixThrows [failed again](http://aspnetci/viewLog.html?buildId=518638&buildTypeId=XPlat_Windows_Win2008r2_Universe).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
ClosingConnectionMidChunkPrefixThrows [failed again](http://aspnetci/viewLog.html?buildId=518634&buildTypeId=XPlat_Windows_Win7_Universe).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
ClosingConnectionMidChunkPrefixThrows [failed again](http://aspnetci/viewLog.html?buildId=519050&buildTypeId=XPlat_Windows_Win2008r2_Universe).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
ClosingConnectionMidChunkPrefixThrows [failed again](http://aspnetci/viewLog.html?buildId=519046&buildTypeId=XPlat_Windows_Win7_Universe).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
DoesNotThrowOnFin [failed again](http://aspnetci/viewLog.html?buildId=519471&buildTypeId=XPlat_Mac_MacUniverseCoreCLR).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
DoesNotThrowOnFin [failed again](http://aspnetci/viewLog.html?buildId=519358&buildTypeId=XPlat_Windows_Win8_Universe).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
DoesNotThrowOnFin [failed again](http://aspnetci/viewLog.html?buildId=519360&buildTypeId=XPlat_Windows_Win10Universe).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
DoesNotThrowOnFin [failed again](http://aspnetci/viewLog.html?buildId=519292&buildTypeId=XPlat_HighSierraOsxUniverse).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
DoesNotThrowOnFin [failed again](http://aspnetci/viewLog.html?buildId=519458&buildTypeId=XPlat_HighSierraOsxUniverse).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
DoesNotThrowOnFin [failed again](http://aspnetci/viewLog.html?buildId=519452&buildTypeId=XPlat_Windows_Win8_Universe).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
ClosingConnectionMidChunkPrefixThrows [failed again](http://aspnetci/viewLog.html?buildId=519454&buildTypeId=XPlat_Windows_Win10Universe).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
ClosingConnectionMidChunkPrefixThrows [failed again](http://aspnetci/viewLog.html?buildId=519572&buildTypeId=Lite_UniverseTest).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
ClosingConnectionMidChunkPrefixThrows [failed with about the same error](http://aspnetci/viewLog.html?buildId=520895&buildTypeId=Lite_UniverseTest).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
ClosingConnectionMidChunkPrefixThrows [failed with about the same error](http://aspnetci/viewLog.html?buildId=522188&buildTypeId=Lite_UniverseTest).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
DoesNotThrowOnFin [failed with about the same error](http://aspnetci/viewLog.html?buildId=523276&buildTypeId=XPlat_Mac_MacUniverseCoreCLR).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
ClosingConnectionMidChunkPrefixThrows [failed with about the same error](http://aspnetci/viewLog.html?buildId=524454&buildTypeId=XPlat_Mac_MacUniverseCoreCLR).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
ClosingConnectionMidChunkPrefixThrows [failed with about the same error](http://aspnetci/viewLog.html?buildId=525077&buildTypeId=XPlat_Windows_Win2008r2_Universe).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by halter73** [comment]
Thes failures should be fixed by dotnet/corefx#31685.

**Comment by JunTaoLuo** [comment]
These tests have not failed for several weeks, shortly after the fix linked was merged. Closing as it's likely fixed.
